core/tests/protractor/privileges.js

```diff
@@ -46,10 +46,18 @@ describe('Permissions for private explorations', function() {
 
       users.login('eve@privileges.com');
       general.openEditor(explorationId);
+
       // Eve is redirected to the homepage.
-      expect(browser.getCurrentUrl()).toEqual(
-        general.SERVER_URL_PREFIX + '/dashboard');
-      users.logout();
+      browser.getCurrentUrl().then(function(curUrl) {
+        var dashboardURL = general.SERVER_URL_PREFIX + '/dashboard';
+        var libraryURL = general.SERVER_URL_PREFIX + '/library';
+
+        // Homepage can be either dashboard or library dependin on learner
+        // Status
+        var atHomeFlag = (curUrl === libraryURL) || (curUrl === dashboardURL);
```

Comment:
This test is meant to check privileges, so don't overload it; it's not meant to check correct redirection.

I think it's fine to check here that Eve is on the library page. You could perhaps add the check for the dashboard page to some other protractor test that involves creating an exploration, though I'd expect there might be a test that already covers that. Please double-check, though.